cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -1375,7 +1368,7 @@ private void deleteAppVersion(ApplicationId appId, ApplicationSpecification spec
     for (WorkflowSpecification workflowSpec : spec.getWorkflows().values()) {
       scheduler.modifySchedulesTriggeredByDeletedProgram(appId.workflow(workflowSpec.getName()));
     }
-    store.removeApplication(appId);
+    store.removeApplication(appId.getAppReference());
```

Comment:
this method is named deleteAppVersion(), it should be deleting a specific version and not all versions.

Though I don't know why it is deleting schedules above, seems like something that was missed earlier?

Reply:
We won't be supporting this function when the FF is turned on - as deleting specific versions is not allowed. Do we still want to retain `removeApplication(appId)` ?

Reply:
We have to retain it if we're still supporting the feature flag as it will still be possible for somebody to call the REST API to delete an app version with the feature flag turned off. When that happens, it shouldn't delete all versions of the app.

If we're completely removing that REST API, then we can remove the `removeApplication(appId)` method, but then it should be done in the same PR that the REST API is removed.